### ISSUE TITLE
Using pretrained weights but getting poor metrics?

### ISSUE THREAD

**Issue by hduxiao** [DoctorKey/C-3PO#5]
Hello, I use the pre-trained weight "resnet18_id_4_deeplabv3_VL_CMU_CD.pth" to run the shell command "resnet18_mtf_id_msf4_deeplabv3_cmu", but the result is very bad, the following is my running log and results, can you help me find the reason?

>PS C:\Users\TEST\Desktop\C-3PO-main>  c:; cd 'c:\Users\TEST\Desktop\C-3PO-main'; & 'C:\Users\TEST\anaconda3\envs\pc\python.exe' 'c:\Users\TEST\.vscode\extensions\ms-python.python-2023.4.1\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher' '52990' '--' 'C:\Users\TEST\Desktop\C-3PO-main/src/train.py' '--test-only' '--save-imgs' '--model' 'vgg16bn_mtf_msf_deeplabv3' '--mtf' 'id' '--msf' '4' '--train-dataset' 'VL_CMU_CD' '--test-dataset' 'VL_CMU_CD' '--input-size' '512' '--resume' 'C:\Users\TEST\Downloads\vgg16bn_id_4_deeplabv3_VL_CMU_CD.pth' 
Not using distributed mode
Namespace(train_dataset='VL_CMU_CD', test_dataset='VL_CMU_CD', test_dataset2='', input_size=512, randomflip=0.5, randomrotate=False, randomcrop=False, data_cv=0, model='vgg16bn_mtf_msf_deeplabv3', mtf='id', msf=4, device='cuda', batch_size=4, epochs=100, workers=16, loss='bi', loss_weight=False, opt='adam', lr_scheduler='cosine', lr=0.0001, warmup=False, momentum=0.9, weight_decay=0, print_freq=50, resume='C:\\Users\\TEST\\Downloads\\vgg16bn_id_4_deeplabv3_VL_CMU_CD.pth', pretrained='', start_epoch=0, eval_every=1, test_only=True, save_imgs=True, save_local=False, world_size=1, dist_url='env://', output_dir='C:/Users/TEST/Desktop/output\\vgg16bn_mtf_msf_deeplabv3_VL_CMU_CD_0/2023-03-13_19-29-23', distributed=False)
Train Aug:
DATA AUG: resize (512, 512)
DATA AUG: random flip 0.5
VL_CMU_CD train: 1008
Test Aug:
DATA AUG: resize (512, 512)
VL_CMU_CD test: 354
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torch\utils\data\dataloader.py:554: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 8 (`cpuset` is not taken into account), which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
MSF: [128, 256, 512, 512]
MTF: mode: id kernel_size: 3
MTF: mode: id kernel_size: 3
MTF: mode: id kernel_size: 3
MTF: mode: id kernel_size: 3
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\models\_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed 
in the future, please use 'weights' instead.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\models\_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=VGG16_BN_Weights.IMAGENET1K_V1`. You can also use `weights=VGG16_BN_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Loss: bi
LR Scheduler: cosine
load from: C:\Users\TEST\Downloads\vgg16bn_id_4_deeplabv3_VL_CMU_CD.pth
load ret: <All keys matched successfully>
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torch\utils\data\dataloader.py:554: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 8 (`cpuset` is not taken into account), which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
C:\Users\TEST\anaconda3\envs\pc\lib\site-packages\torchvision\transforms\functional.py:442: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
Test:  [  0/354]  eta: 5:26:46  Prec: 0.999 (0.999)  Rec: 0.034 (0.034)  Acc: 0.092 (0.092)  F1score: 0.0656 (0.0656)  time: 55.3849  data: 53.1827  max mem: 717       
Test:  [100/354]  eta: 0:03:54  Prec: 1.000 (0.994)  Rec: 0.188 (0.113)  Acc: 0.269 (0.250)  F1score: 0.3163 (0.1955)  time: 0.3864  data: 0.0014  max mem: 721
Test:  [200/354]  eta: 0:01:39  Prec: 0.995 (0.996)  Rec: 0.115 (0.111)  Acc: 0.157 (0.236)  F1score: 0.2069 (0.1886)  time: 0.3468  data: 0.0016  max mem: 721
Test:  [300/354]  eta: 0:00:30  Prec: 0.994 (0.981)  Rec: 0.127 (0.104)  Acc: 0.169 (0.214)  F1score: 0.2245 (0.1784)  time: 0.4156  data: 0.0014  max mem: 721
Test: Total time: 0:03:13
Test:  Total: 354 Metric Prec: 0.9834 Recall: 0.0997 F1: 0.1728
tensor(0.1728)

**Comment by DoctorKey** [comment]
1. Add `--save-imgs` in your command to obtain visualization results.
2. Check the dataset. I think you may use a wrong dataset.

```
Train Aug:
DATA AUG: resize (512, 512)
DATA AUG: random flip 0.5
VL_CMU_CD train: 3732
Test Aug:
DATA AUG: resize (512, 512)
VL_CMU_CD test: 429

Test:  [  0/429]  eta: 0:21:19  Prec: 0.906 (0.906)  Rec: 0.950 (0.950)  Acc: 0.997 (0.997)  F1score: 0.9272 (0.9272)  time: 2.9831  data: 1.0247  max mem: 759
Test:  [100/429]  eta: 0:00:20  Prec: 0.881 (0.836)  Rec: 0.970 (0.775)  Acc: 0.974 (0.980)  F1score: 0.9234 (0.7636)  time: 0.0314  data: 0.0022  max mem: 763
Test:  [200/429]  eta: 0:00:10  Prec: 0.912 (0.841)  Rec: 0.948 (0.796)  Acc: 0.999 (0.979)  F1score: 0.9296 (0.7800)  time: 0.0336  data: 0.0027  max mem: 763
Test:  [300/429]  eta: 0:00:05  Prec: 0.945 (0.841)  Rec: 0.960 (0.792)  Acc: 0.999 (0.978)  F1score: 0.9525 (0.7801)  time: 0.0313  data: 0.0022  max mem: 763
Test:  [400/429]  eta: 0:00:01  Prec: 0.908 (0.836)  Rec: 0.996 (0.784)  Acc: 0.998 (0.978)  F1score: 0.9500 (0.7742)  time: 0.0317  data: 0.0022  max mem: 763
Test: Total time: 0:00:16
Test:  Total: 429 Metric Prec: 0.8367 Recall: 0.7893 F1: 0.7782
tensor(0.7782)
```

**Comment by DoctorKey** [comment]
Please check README. I have updated the download link. You can download `VL-CMU-CD-binary255` at [here](https://huggingface.co/datasets/Flourish/VL-CMU-CD/blob/main/VL-CMU-CD-binary255.zip).
